employees/tests/test_unit_report_viewset.py

```diff
@@ -642,3 +643,100 @@ def test_project_report_list_get_queryset_method_should_return_queryset_containi
         self.assertFalse(other_project_report in queryset)
         self.assertEqual(queryset[0], other_report_1)
         self.assertEqual(queryset[1], self.report)
+
+
+class ProjectReportDetailTests(TestCase):
+    def setUp(self):
+        self.user = CustomUser(
+            email="testuser@codepoets.it", password="newuserpasswd", first_name="John", last_name="Doe", country="PL"
+        )
+        self.user.full_clean()
+        self.user.save()
+
+        self.project = Project(name="Test Project", start_date=datetime.datetime.now())
+        self.project.full_clean()
+        self.project.save()
+
+        self.report = Report(
+            date=datetime.datetime.now().date(),
+            description="Some description",
+            author=self.user,
+            project=self.project,
+            work_hours=Decimal("8.00"),
+        )
+        self.report.full_clean()
+        self.report.save()
+
+    def test_project_report_detail_view_should_display_report_details(self):
+        request = APIRequestFactory().get(path=reverse("project-report-detail", args=(self.report.pk,)))
```

Comment:
Please use `self.client` that is delivered with Django `TestCase`. You can simply make `self.client.force_login(self.user)` to make it authenticated (ie. in `setUp()`). You can take a look how it is used in other refactored tests for Django views.

Reply:
Done